### PR TITLE
Pyzhou/TDI-42451 salesforce bulk support json

### PR DESCRIPTION
**Pull request by pyzhou-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-42451

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/727/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/727/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/728/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/728/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/750/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/750/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/752/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/752/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/753/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/753/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/755/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/755/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/756/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/756/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/757/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/757/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/759/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/759/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/760/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/760/)

File: components/components-salesforce/components-salesforce-definition/src/main/java/org/talend/components/salesforce/tsalesforcebulkexec/TSalesforceBulkExecProperties.java [hunk 1]
```diff
@@ -43,6 +45,8 @@
 
     public Property<Boolean> outputUpsertKey = newBoolean("outputUpsertKey");
 
+    public Property<ContentType> contentType = new EnumProperty(ContentType.class, "contentType");
```
Comment (jzhao-talend) [inline components/components-salesforce/components-salesforce-definition/src/main/java/org/talend/components/salesforce/tsalesforcebulkexec/TSalesforceBulkExecProperties.java:48]: did you try to hide JSON type for bulkv2 ? not sure we can do it on the UI side.
or we can throw a runtime exception for the customer.

File: components/components-salesforce/components-salesforce-runtime/src/main/resources/org/talend/components/salesforce/runtime/SalesforceBulkRuntime.properties [hunk 2]
```diff
@@ -1 +1,2 @@
-warn.batch.timeout=Your query exceed the 10 minutes limit!
\ No newline at end of file
+warn.batch.timeout=Your query exceed the 10 minutes limit!
+error.bulk.jsonArray=Failed to bulk load the file. Edit the root JSON path to use JSON array and try again.
```
Comment (jzhao-talend) [inline components/components-salesforce/components-salesforce-runtime/src/main/resources/org/talend/components/salesforce/runtime/SalesforceBulkRuntime.properties:2]: As the workflow, don't forget to check the property display value with Hiyam Massy

File: components/components-salesforce/components-salesforce-definition/src/main/java/org/talend/components/salesforce/tsalesforcebulkexec/TSalesforceBulkExecProperties.java (hunk 1 above)
Comment (pyzhou-talend) [reply components/components-salesforce/components-salesforce-definition/src/main/java/org/talend/components/salesforce/tsalesforcebulkexec/TSalesforceBulkExecProperties.java:48]: bulkv2 does not support JSON format. We can only use it in bulk v1.

File: components/components-salesforce/components-salesforce-runtime/src/main/resources/org/talend/components/salesforce/runtime/SalesforceBulkRuntime.properties (hunk 2 above)
Comment (pyzhou-talend) [reply components/components-salesforce/components-salesforce-runtime/src/main/resources/org/talend/components/salesforce/runtime/SalesforceBulkRuntime.properties:2]: Yes. This is the statement that Hiyam Massy gives me.

File: components/components-salesforce/components-salesforce-definition/src/main/java/org/talend/components/salesforce/tsalesforcebulkexec/TSalesforceBulkExecProperties.java (hunk 1 above)
Comment (pyzhou-talend) [reply components/components-salesforce/components-salesforce-definition/src/main/java/org/talend/components/salesforce/tsalesforcebulkexec/TSalesforceBulkExecProperties.java:48]: Thanks, @jzhao-talend . I have modified the code to hide ContentType for bulkV2.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/762/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/762/)

**Review by jzhao-talend — APPROVED** [review]
LGTM

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/777/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/777/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/778/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/778/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/779/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/779/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/780/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/780/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/782/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/782/)

**Pull request merged**